public/app/plugins/datasource/prometheus/datasource.tsx

```diff
@@ -811,6 +824,71 @@ export class PrometheusDatasource
     return result?.data?.data?.map((value: any) => ({ text: value })) ?? [];
   }
 
+  async getBuildInfo() {
+    try {
+      const buildInfo = await fetchDataSourceBuildInfo(this);
+      return buildInfo;
+    } catch (error) {
+      // We don't want to break the rest of functionality if build info does not work correctly
+      return undefined;
+    }
+  }
+
+  getBuildInfoMessage(buildInfo: PromBuildInfo) {
+    const enabled = <Badge color="green" icon="check" text="Ruler API enabled" />;
+    const disabled = <Badge color="orange" icon="exclamation-triangle" text="Ruler API not enabled" />;
+
+    const LOGOS = {
+      [PromApplication.Cortex]: '/public/app/plugins/datasource/prometheus/img/cortex_logo.svg',
+      [PromApplication.Mimir]: '/public/app/plugins/datasource/prometheus/img/mimir_logo.svg',
+      [PromApplication.Prometheus]: '/public/app/plugins/datasource/prometheus/img/prometheus_logo.svg',
+    };
+
+    const COLORS: Record<PromApplication, BadgeColor> = {
+      [PromApplication.Cortex]: 'blue',
+      [PromApplication.Mimir]: 'orange',
+      [PromApplication.Prometheus]: 'red',
+    };
+
+    // this will inform the user about what "subtype" the datasource is; Mimir, Cortex or vanilla Prometheus
+    const applicationSubType = (
+      <Badge
+        text={
+          <span>
+            <img
+              style={{ width: 14, height: 14, verticalAlign: 'text-bottom' }}
+              src={LOGOS[buildInfo.application ?? PromApplication.Prometheus]}
+            />{' '}
+            {buildInfo.application}
+          </span>
+        }
+        color={COLORS[buildInfo.application ?? PromApplication.Prometheus]}
+      />
+    );
+
+    return (
+      <div
+        style={{
+          display: 'grid',
+          gridTemplateColumns: 'max-content max-content',
+          rowGap: '0.5rem',
+          columnGap: '2rem',
+          marginTop: '1rem',
+        }}
+      >
+        <div>Type</div>
+        <div>{applicationSubType}</div>
+        {/* Prometheus does not have a Ruler API – so skip for that type */}
```

Comment:
Don't we want to tell Prometheus users that some functionality won't be available for them in that case?

Reply:
Good suggestion, I think we should try to include that. We've seen some instances where customers are confused about not being able to manage rules with their vanilla Prometheus datasource.

Reply:
Added an "unsupported" badge and a tooltip to clarify to the user that we do not support rule management with Prometheus.

<img width="510" alt="image" src="https://user-images.githubusercontent.com/868844/161514685-2086e1d7-29a9-45e1-95fb-7a185a4868cc.png">

Reply:
Awesome! 🚀